### PR TITLE
PodStatusUtil#isRunning & #isInitializing (#2293)

### LINKED ISSUE #2293: Should report if a pod is "running"
Kubectl, oc and the web console all implement an almost identical logic to determine if a pod is considered "running".

* Here's the logic in the oc command line: https://github.com/openshift/origin/blob/master/vendor/k8s.io/kubernetes/pkg/printers/internalversion/printers.go#L695-L781
* Here's the almost identical logic in the openshift web-console: https://github.com/openshift/origin-web-console/blob/master/app/scripts/filters/resources.js#L1012-L1088

I have an [implementation](https://github.com/redhat-developer/intellij-kubernetes/blob/master/src/main/kotlin/org/jboss/tools/intellij/kubernetes/model/util/PodStatus.kt) that I'd be willing to contribute to the kubernetes-client so that 3rd party users of the client could reuse it instead replicating the logic.

The most appropriate location for it seems the package `io.fabric8.kubernetes.client.utils`. There's a similar utility at `io.fabric8.kubernetes.client.internal.readiness.Readiness` but it's in an internal package that's obviously not intended for external consumption.

Thoughts?

### PR DESCRIPTION
**Pull request by adietish**
fixes #2293 

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [x] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [ ] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] Code contributed by me aligns with current project license: [Apache 2.0](https://www.apache.org/licenses/LICENSE-2.0)
 - [ ] I Added [CHANGELOG](../CHANGELOG.md) entry regarding this change
 - [x] I have implemented unit tests to cover my changes
 - [x] I have added/updated the [javadocs](https://www.javadoc.io/doc/io.fabric8/kubernetes-client/latest/index.html) and other [documentation](https://github.com/fabric8io/kubernetes-client/blob/master/doc/CHEATSHEET.md) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=fabric8io_kubernetes-client) report
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

**Comment by rohanKanojia** [comment]
E2E test failures are unrelated to this change. I've created #2553 to address these failures

**Comment by rohanKanojia** [comment]
@adietish : Could you please rebase with master whenever you get time to revisit this? I think tests should pass now. Apologies for inconvenience ..

**Comment by adietish** [comment]
@rohanKanojia thx! rebased.

**Comment by adietish** [comment]
@rohanKanojia javadoc failure is afaics not in my code. Guess one can thus interpret all checks as green as for this PR.

**Comment by rohanKanojia** [comment]
Umm, I checked logs and looks like it's complaining for this line:
```
2020-10-21T12:24:18.3003398Z [ERROR] /home/runner/work/kubernetes-client/kubernetes-client/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/utils/PodStatusUtil.java:40: error: unexpected text
2020-10-21T12:24:18.3007610Z [ERROR]      * {@link <a href="https://github.com/openshift/origin/blob/master/vendor/k8s.io/api/core/v1/types.go#L3564"/>}
```

**Comment by adietish** [comment]
> Umm, I checked logs and looks like it's complaining for this line:
> 
> ```
> 2020-10-21T12:24:18.3003398Z [ERROR] /home/runner/work/kubernetes-client/kubernetes-client/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/utils/PodStatusUtil.java:40: error: unexpected text
> 2020-10-21T12:24:18.3007610Z [ERROR]      * {@link <a href="https://github.com/openshift/origin/blob/master/vendor/k8s.io/api/core/v1/types.go#L3564"/>}
> ```

what's the problem with that? Looks valid to me, what am I missing?

**Comment by adietish** [comment]
Removed the braced @link tag. Hopes this helps

File: kubernetes-client/src/test/java/io/fabric8/kubernetes/client/utils/PodStatusUtilTest.java [hunk 1]
```diff
@@ -0,0 +1,360 @@
...
+            null,
+            null,
+            containerStateRunning()
+        )
+    );
+
+    @Test
+    public void isRunning_should_return_true_if_pod_is_in_phase_running() {
+        // given
+        Pod pod = pod("starwars")
```
Comment (manusa) [inline kubernetes-client/src/test/java/io/fabric8/kubernetes/client/utils/PodStatusUtilTest.java:49]: :heart: so clear and legible

**Review by manusa — APPROVED** [review]

File: kubernetes-client/src/test/java/io/fabric8/kubernetes/client/utils/PodMockUtils.java [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+    }
+
+    public static ContainerStateWaiting containerStateWaiting(String reason) {
+        ContainerStateWaiting mock = mock(ContainerStateWaiting.class);
+        when(mock.getReason())
+                .thenReturn(reason);
+        return mock;
+    }
+
+    public static ContainerStateRunning containerStateRunning() {
```
Comment (rohanKanojia) [inline kubernetes-client/src/test/java/io/fabric8/kubernetes/client/utils/PodMockUtils.java:92]: Just a minor comment, Could you please add some javadocs for this class?

Comment (adietish) [reply kubernetes-client/src/test/java/io/fabric8/kubernetes/client/utils/PodMockUtils.java:168]: This is a class that holds utils to create a pod mock. I added extensive javadocs.

**Review by rohanKanojia — APPROVED** [review]

**Comment by adietish** [comment]
@rohanKanojia, @manusa, @oscerd  thanks for the approval! Can we now hit the big green button, please?

**Comment by rohanKanojia** [comment]
[merge]

**Pull request merged**